bears/vcs/git/GitCommitBear.py

```diff
@@ -110,6 +114,38 @@ def check_shortlog(self,
                     self,
                     "Shortlog of HEAD commit does not match given regex.")
 
+        if shortlog_imperative_check:
+            colon_pos = shortlog.find(':')
+            shortlog = shortlog[colon_pos:] if colon_pos != -1 else shortlog
+            has_flaws = self.check_imperative(shortlog)
+            if has_flaws:
+                bad_words = ', '.join([words[0] for words in has_flaws])
+                yield Result(self,
+                             "Shortlog of HEAD commit isn't imperative mood,"
+                             "bad words are '{}'".format(bad_words))
+
+    @staticmethod
+    def check_imperative(paragraph):
+        """
+        Check the given sentence/s for Imperatives.
```

Comment:
You meant something like `["whoa", "added some stuff"]` ? Yes that'd work
too
ᐧ

On Mon, Apr 4, 2016 at 11:37 PM, Mischa Krüger <notifications@github.com>
wrote:

> In bears/vcs/git/GitCommitBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/315#discussion_r58420033>
> :
>
> > @@ -110,6 +114,38 @@ def check_shortlog(self,
> >                      self,
> >                      "Shortlog of HEAD commit does not match given regex.")
> >
> > +        if shortlog_imperative_check:
> > +            colon_pos = shortlog.find(':')
> > +            shortlog = shortlog[colon_pos:] if colon_pos != -1 else shortlog
> > +            has_flaws = self.check_imperative(shortlog)
> > +            if has_flaws:
> > +                bad_words = ', '.join([words[0] for words in has_flaws])
> > +                yield Result(self,
> > +                             "Shortlog of HEAD commit isn't imperative mood,"
> > +                             "bad words are '{}'".format(bad_words))
> > +
> > +    @staticmethod
> > +    def check_imperative(paragraph):
> > +        """
> > +        Check the given sentence/s for Imperatives.
>
> You say sentence*s*. Do you accept sequences of strings?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/315/files/23d9af66bcb498b19adffe8effb5af1ec8aa4f40#r58420033>
>